core/tx_pool.go

```diff
@@ -1643,6 +1685,41 @@ func (pool *TxPool) demoteUnexecutables() {
 	}
 }
 
+// insertTxToLifecycle insert the tx to lifecycle calculator
+func (pool *TxPool) insertTxToLifecycle(tx common.Hash) {
+	if !pool.enableTxLifecycles {
+		return
+	}
+
+	pool.txLifecycles.Add(tx, uint64(time.Now().Unix()))
+}
+
+// calculateTxLifecycle calculate the txs lifecycle
+func (pool *TxPool) calculateTxsLifecycle(txs []common.Hash, t uint64) {
+	if !pool.enableTxLifecycles {
+		return
+	}
+
+	var (
+		totalTxsLifecycle uint64
+		txCount           uint64
+	)
+	for _, tx := range txs {
+		addTime, ok := pool.txLifecycles.Get(tx)
+		if !ok {
+			continue
+		}
+		if t > addTime.(uint64) {
+			txCount++
+			txLifecycle := t - addTime.(uint64)
+			totalTxsLifecycle += txLifecycle
+		}
+	}
+
+	median := float64(totalTxsLifecycle) / float64(txCount)
+	txLifecycleTimer.Update(time.Duration(median) * time.Second)
```

Comment:
Timer can already calculate a global mean for us. I think we can get away with calling timer.Update() in the for loop above with `txLifecycle`

Reply:
ok